src/Runtime/Core/Collections/GenerationalDictionary.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.ML.Probabilistic.Core.Collections
+{
+    using System;
+
+    /// <summary>
+    /// Hash table implementation that has a constant-time Clear() operation.
+    /// Caveat is that it does not set values to `default` values, so it should not be used
```

Comment:
This comment is confusing.  Do you mean to say that the table may retain references to cleared values, preventing garbage collection of those values until the GenerationalDictionary is collected?

Reply:
Seems so from the implementation.
@lazy Shouldn't that restriction on `TValue` be enforced then?

Reply:
> Shouldn't that restriction on TValue be enforced then?
No. Struct (which is a value type) can contain a field which is a reference itself.
I will clarify the comment.

Reply:
Can we use `where TValue : unmanaged` with netstandard2.0? My VS seems fine with it and points to TarjanStateInfo being a class.

Reply:
I have removed the "caveat" comment and added the unmanaged restriction. Is that fine?

Reply:
Yes

Reply:
Fine with me.